pkg/kubelet/winstats/winstats.go

```diff
@@ -0,0 +1,203 @@
+
+	err = <-errChan
+	return client, err
+}
+
+// startNodeMonitoring starts reading perf counters of the node and updates
+// the client struct with cpu and memory stats
+func (c *Client) startNodeMonitoring(errChan chan error) {
+	cpuChan, err := readPerformanceCounter(cpuQuery)
+
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	memWorkingSetChan, err := readPerformanceCounter(memoryPrivWorkingSetQuery)
+
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	memCommittedBytesChan, err := readPerformanceCounter(memoryCommittedBytesQuery)
+
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	// no error, send nil over channel
+	errChan <- nil
+
+	for {
+		select {
+		case cpu := <-cpuChan:
+			c.updateCPU(cpu)
+		case mWorkingSet := <-memWorkingSetChan:
+			c.updateMemoryWorkingSet(mWorkingSet)
+		case mCommittedBytes := <-memCommittedBytesChan:
+			c.updateMemoryCommittedBytes(mCommittedBytes)
+		}
+	}
+}
+
+func (c *Client) updateCPU(cpu Metric) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	cpuCores := runtime.NumCPU()
+	// This converts perf counter data which is cpu percentage for all cores into nanoseconds.
+	// The formula is (cpuPercentage / 100.0) * #cores * 1e+9 (nano seconds). More info here:
+	// https://github.com/kubernetes/heapster/issues/650
+	c.cpuUsageCoreNanoSeconds += uint64((cpu.Value / 100.0) * float64(cpuCores) * 1000000000)
+}
+
+func (c *Client) updateMemoryWorkingSet(mWorkingSet Metric) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.memoryPrivWorkingSetBytes = uint64(mWorkingSet.Value)
+}
+
+func (c *Client) updateMemoryCommittedBytes(mCommittedBytes Metric) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.memoryCommittedBytes = uint64(mCommittedBytes.Value)
+}
+
+// WinContainerInfos returns a map of container infos. The map contains node and
+// pod level stats. Analogous to cadvisor GetContainerInfoV2 method.
+func (c *Client) WinContainerInfos() (map[string]cadvisorapiv2.ContainerInfo, error) {
+	infos := make(map[string]cadvisorapiv2.ContainerInfo)
+
+	// root (node) container
+	infos["/"] = *c.createRootContainerInfo()
+
+	return infos, nil
+}
+
+// WinMachineInfo returns a cadvisorapi.MachineInfo with details about the
+// node machine. Analogous to cadvisor MachineInfo method.
+func (c *Client) WinMachineInfo() (*cadvisorapi.MachineInfo, error) {
+	hostname, err := os.Hostname()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &cadvisorapi.MachineInfo{
+		NumCores:       runtime.NumCPU(),
+		MemoryCapacity: c.memoryPhysicalCapacityBytes,
+		MachineID:      hostname,
+	}, nil
+}
+
+// WinVersionInfo returns a  cadvisorapi.VersionInfo with version info of
+// the kernel and docker runtime. Analogous to cadvisor VersionInfo method.
+func (c *Client) WinVersionInfo() (*cadvisorapi.VersionInfo, error) {
+	dockerServerVersion, err := c.dockerClient.ServerVersion(context.Background())
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &cadvisorapi.VersionInfo{
+		KernelVersion:    dockerServerVersion.KernelVersion,
```

Comment:
Fixed. In my latest commit as I removed Docker client and used `exec` to get kernel version instead.